### ISSUE TITLE
Bug when filtering overlong data with tool configurations

### ISSUE THREAD

**Issue by huaiyizhao** [verl-project/verl#3936]
### System Info

----------Python Info----------
Version      : 3.12.3
Compiler     : GCC 13.3.0
Build        : ('main', 'Feb  4 2025 14:48:35')
Arch         : ('64bit', 'ELF')
------------Pip Info-----------
Version      : 25.2
Directory    : /usr/local/lib/python3.12/dist-packages/pip
vllm         : not found.
sglang       : 0.5.3
ray          : 2.49.2
torch        : 2.8.0
----------verl Info-----------
Version      : 0.7.0.dev
Directory    : /app/verl/verl
Commit Hash  : ea1918bda6402358aef11e52d3faad2adacb1d81
----------Platform Info----------
Platform     : Linux-5.4.241-1-tlinux4-0017.7-x86_64-with-glibc2.39
system       : Linux
node         : TENCENT64.site
release      : 5.4.241-1-tlinux4-0017.7
version      : #1 SMP Thu Jan 18 11:33:00 CST 2024
----------Environment----------
CUDA Runtime : 12.8
CUDA Compiler : Cuda compilation tools, release 12.8, V12.8.93
----------System Info----------
CPU Memory      : 2265.25 GB
GPU Count       : 8
GPU 1   Type    : NVIDIA H20
GPU 1   Memory  : 95.58 GB
GPU 2   Type    : NVIDIA H20
GPU 2   Memory  : 95.58 GB
GPU 3   Type    : NVIDIA H20
GPU 3   Memory  : 95.58 GB
GPU 4   Type    : NVIDIA H20
GPU 4   Memory  : 95.58 GB
GPU 5   Type    : NVIDIA H20
GPU 5   Memory  : 95.58 GB
GPU 6   Type    : NVIDIA H20
GPU 6   Memory  : 95.58 GB
GPU 7   Type    : NVIDIA H20
GPU 7   Memory  : 95.58 GB
GPU 8   Type    : NVIDIA H20
GPU 8   Memory  : 95.58 GB

### Information

- [x] The official example scripts
- [ ] My own modified scripts

### Tasks

- [x] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

The maybe_filter_out_long_prompts in rl_dataset.py cannot see tool configurations, so it will apply_chat_template without tools. 
If `len(sample) < max_prompt_length < len(sample) + len(tool related prompts)`, It will not be filtered. When it goes to _handle_pending_state in tool_agent_loop.py, it will generated prompt_ids with tool related prompts, which > `max_prompt_length`. Then, the padding on agent_loop.py:_run_agent_loop will fail on this sample, which, eventually causes error on _postprocess where torch.cat expect tensors of same length for prompt_ids ([1, max_prompt_length])

### Expected behavior

filter out long samples with tools considered

**Comment by alekseymalakhov11** [comment]
Hi 👋
This issue has already been addressed and will be resolved in #3867

**Comment by Kirrito-k423** [comment]
中文：这个问题看起来已经由已合并的 PR #3950 修复。#3936 报告的问题是数据过滤阶段没有把 tool configuration/tool schema 计入 chat template 长度，导致后续 tool agent rollout 生成的 prompt_ids 超过 max_prompt_length。#3950 已在 2025-11-03 合并，并明确说明 rl_dataset 现在会在带工具描述应用 chat template 后再判断 prompt length。当前 main 中也能看到 data.tool_config_path/function_tool_path 会传到 RLHFDataset，并在 tokenizer/processor 的 apply_chat_template 过滤逻辑中传入 tools。

English: This appears to have been fixed by merged PR #3950. The reported problem was that dataset-side overlong prompt filtering did not include the tool configuration/tool schemas in the chat template length, while the tool-agent rollout later did include them and could exceed max_prompt_length. PR #3950 was merged on 2025-11-03 and explicitly states that rl_dataset now filters after applying the chat template with tool descriptions. Current main also shows data.tool_config_path/function_tool_path being passed into RLHFDataset and tools being included in tokenizer/processor apply_chat_template during prompt-length filtering.

This response was generated by a scheduled Codex GPT5.5 issue processing task.
